### DISCUSSION TITLE
Output a "flat" YAML

### DISCUSSION THREAD

**Discussion by debuglevel** [mikefarah/yq#700] [Q&A] [answered]
**Please describe your feature request.**
Create an output with the full path of each key instead of the usual hierarchical YAML syntax. I don't know whether this would actually be a valid YAML, but it would be nice to be used for further processing.

**Describe the solution you'd like**
Example given:
```yaml
animals:
  dogs: 3
  cats: 5
```
would convert it to
```yaml
animals.dogs: 3
animals.cats: 3
```

This would be useful for further processing, e.g. creating environment variables like often used with for Docker:
```bash
cat settings.yml | yq eval flatten-my-stuff | ./some-bash-script.sh

ANIMALS_DOG=3
ANIMALS_CATS=5
```

**Describe alternatives you've considered**
Probably could also be done with some Python.

**Comment by mikefarah** [comment, marked as answer]
```
.animals.[] | {(path | join(".")): .}
```

**Reply by mikefarah** [reply]
Also, see https://github.com/mikefarah/yq/discussions/699 for a more generic approach

**Reply by pascal-hofmann** [reply]
I need the reverse. Is there a way to unflatten a yaml file with yq?

**Reply by mikefarah** [reply]
Sure, please see https://mikefarah.gitbook.io/yq/usage/properties#encode-properties (or raise a separate discussion if your use case is different
